rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -896,10 +1168,12 @@ extern "C"
 
         waitset->fixed_guard_conditions = fixed_guard_conditions;
         if (fixed_guard_conditions) {
+/*
```

Comment:
This check should definitely not be remove. But currently the check is wrong. It is fine that `guard_conditions` is `0` if and only if `guard_condition_count` is zero.